lib/Sema/TypeCheckConcurrency.h

```diff
@@ -280,12 +280,17 @@ enum class FunctionCheckKind {
 /// \param funcCheckKind Describes whether function types in this reference
 /// should be checked for sendability of their results, params, or both
 ///
+/// \param topLevelLoc Provides an alternative source location to `loc`
+/// to be used for reporting the top level diagnostic while auxiliary
+/// warnings and diagnostics are reported at `loc`.
+///
 /// \returns true if an problem was detected, false otherwise.
 bool diagnoseNonSendableTypesInReference(
     ConcreteDeclRef declRef, const DeclContext *fromDC, SourceLoc loc,
     SendableCheckReason refKind,
     Optional<ActorIsolation> knownIsolation = None,
-    FunctionCheckKind funcCheckKind = FunctionCheckKind::ParamsResults);
+    FunctionCheckKind funcCheckKind = FunctionCheckKind::ParamsResults,
+    Optional<SourceLoc> topLevelLoc = None);
```

Comment:
`SourceLoc` already has a null/none value representation via the zero-arge constructor `SourceLoc()` (see `SourceLoc::isNull()`), so that would be preferred over `Optional<SourceLoc>`.